azure_costs_exporter/views.py

```diff
@@ -11,14 +12,19 @@ def health():
 @bp.route("/metrics", methods=['GET'])
 def metrics():
 
+    registry = CollectorRegistry()
+
     try:
-        metrics = query_metrics(current_app.config['ENROLLMENT_NUMBER'],
-                                current_app.config['BILLING_API_ACCESS_KEY'],
-                                current_app.config.get('PROMETHEUS_METRIC_NAME', 'azure_costs')
+        query_metrics(registry,
+                      current_app.config['ENROLLMENT_NUMBER'],
+                      current_app.config['BILLING_API_ACCESS_KEY'],
+                      current_app.config.get('PROMETHEUS_METRIC_NAME', 'azure_costs')
                                 )
     except Exception as e:
         abort(Response("Scrape failed: {}".format(e),
                        status=502)
               )
 
-    return metrics, 200, {'Content-Type': 'text/plain; charset=utf-8'}
\ No newline at end of file
+    content = generate_latest(registry)
```

Comment:
To stick to the norm, you will have to invert the control flow to something like:

```
REGISTRY.register(AzureCollector(
    current_app.config['ENROLLMENT_NUMBER'],
    current_app.config['BILLING_API_ACCESS_KEY'],
    current_app.config.get('PROMETHEUS_METRIC_NAME', 'azure_costs')))

@bp.route("/metrics", methods=['GET'])
def metrics():
    try:
        content = generate_latest()
        return content, 200, {'Content-Type': CONTENT_TYPE_LATEST}
    except Exception as e:
         log.exception("Scrape failed")
         abort(Response("Scrape failed: {}".format(e), status=502)))
```

This implies:

* you register your AzureCollector once at startup in the default `REGISTRY`
* the `generate_latest()` internally loops over the registry and calls your custom collector
* within the `AzureCollector.collect()` you will then query Azure, translate the received data, and yield the appropriate Prometheus metrics.